### PR TITLE
Test connection and rbac for HoH scenario cases

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>

Fixed: https://github.com/stolostron/backlog/issues/24008

### CODE REVIEW

File: test/setup/microshift/microshift_setup.sh [hunk 1]
```diff
@@ -6,16 +6,23 @@ echo "currentDir: $currentDir"
 if [[ -z "$(docker ps | grep "${clusterName}")" ]]; then
   echo "Creating microshift cluster ${clusterName}..."
   docker run -d --rm --name $clusterName --privileged -v hub-data:/var/lib -p 127.0.0.1:6443:6443 -p 127.0.0.1:30095:9095 -p 127.0.0.1:30080:8080 quay.io/microshift/microshift-aio:latest
-  sleep 5
-  until [ $(docker inspect -f {{.State.Running}} ${clusterName})=="true" ]; do
-    echo "Waiting for $clusterName cluster to be ready..."
-    sleep 5;
-  done;
-  docker cp $clusterName:/var/lib/microshift/resources/kubeadmin/kubeconfig $currentDir/../config/kubeconfig-microshift
+  until docker cp $clusterName:/var/lib/microshift/resources/kubeadmin/kubeconfig $currentDir/../config/kubeconfig-microshift > /dev/null 2>&1
```
Comment (yanmxa) [inline test/setup/microshift/microshift_setup.sh:9]: The `until docker cp ...` script is from @morvencao

File: test/pkg/test/hoh-client_test.go [hunk 2]
```diff
@@ -17,25 +17,25 @@ import (
 	"github.com/stolostron/hub-of-hubs/test/pkg/utils"
 )
 
-var _ = Describe("Client", Label("connection"), func() {
+var _ = Describe("Check all the clients could connect to the HoH servers", Label("connection"), func() {
```
Comment (morvencao) [inline test/pkg/test/hoh-client_test.go:20]: nit: consider to use `hubClient.Discovery().RESTClient().Get().AbsPath("/healthz").DoRaw()` to check the health of cluster.

Comment (yanmxa) [reply test/pkg/test/hoh-client_test.go:20]: That's a good idea! I added the health check spec in it.

**Review by morvencao — COMMENTED** [review]
/LGTM

File: test/setup/hoh/hub-of-hubs-rbac/secret/role_bindings.yaml [hunk 3]
```diff
@@ -1,4 +1,7 @@
 roleBindings:
   "kube:admin":
+    roles:
+      - admin
+  "system:masters":
```
Comment (clyang82) [inline test/setup/hoh/hub-of-hubs-rbac/secret/role_bindings.yaml:5]: We can remove `kube:admin` since we do not have such user in microshift context.

**Review by clyang82 — CHANGES_REQUESTED** [review]

**Review by morvencao — COMMENTED** [review]
/lgtm

**Pull request merged**